test/built-ins/RegExp/prototype/Symbol.match/builtin-coerce-global.js

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Type coercion of `global` property value
+es6id: 21.2.5.6
+info: >
+    [...]
+    5. Let global be ToBoolean(Get(rx, "global")).
+    6. ReturnIfAbrupt(global).
+    7. If global is false, then
+       a. Return RegExpExec(rx, S).
+
+    21.2.5.2.1 Runtime Semantics: RegExpExec ( R, S )
+
+    [...]
+    7. Return RegExpBuiltinExec(R, S).
+
+    21.2.5.2.2 Runtime Semantics: RegExpBuiltinExec ( R, S )
+
+    [...]
+    6. Let global be ToBoolean(Get(R, "global")).
+features: [Symbol.match]
+---*/
+
+var r = /./;
+var val, result;
+Object.defineProperty(r, 'global', {
+  get: function() {
+    return val;
+  }
+});
+
+val = false;
+result = r[Symbol.match]('ab');
```

Comment:
Rather than duplicating the call to `@@match` and the assertions, can you create a function that does all these? I see two different outcomes, so we should have two functions.